### ISSUE TITLE
cuckoo fails on most dotnet executables

### ISSUE THREAD

**Issue by overflawed** [cuckoosandbox/cuckoo#1151]
Hi Guys, 

I see there are other related issues, but users seems to have abandoned them and none of them are solved. 
I will try to provide any information needed to help solving this issue.

**Cuckoo analysis fails on almost every dotnet executables with the following error:**
[lib.cuckoo.core.scheduler] ERROR: Analysis failed: The package "modules.packages.exe" start function raised an error: Unable to execute the initial process, analysis aborted.

**Cuckoo/Agents versions tested so far:**
- 2.0-rc1 (2016 Jun ~19th)
- 2.0-rc1 (2016 Oct 31st)
- 2.0-dev (2016 Oct 31st)

**Cuckoo host:** Ubuntu 14.04
**Guest:**  VMWare ESX 6 - Windows 7 Professional x64

**Cuckoo log:**
```
2016-11-01 14:52:36,313 [lib.cuckoo.core.scheduler] DEBUG: Processing task #135
2016-11-01 14:52:36,315 [lib.cuckoo.core.scheduler] INFO: Starting analysis of FILE "dotnet_sample.exe" (task #135, options "route=none")
2016-11-01 14:52:36,341 [lib.cuckoo.core.scheduler] INFO: File already exists at "/opt/cuckoo/storage/binaries/2c0e6d93622ec2bd26ad1a20d0caa09ffc7cb073d40758cb65bdfd45aec8435b"
2016-11-01 14:52:36,392 [lib.cuckoo.core.scheduler] INFO: Task #135: acquired machine 01-win7-x64 (label=01-win7-x64)
2016-11-01 14:52:36,433 [modules.auxiliary.sniffer] INFO: Started sniffer with PID 31091 (interface=eth0, host=172.16.254.51, pcap=/opt/cuckoo/storage/analyses/135/dump.pcap)
2016-11-01 14:52:36,433 [lib.cuckoo.core.plugins] DEBUG: Started auxiliary module: Sniffer
tcpdump: listening on eth0, link-type EN10MB (Ethernet), capture size 65535 bytes
2016-11-01 14:52:36,684 [modules.machinery.vsphere] INFO: Reverting machine 01-win7-x64 to snapshot cuckoo2GB
2016-11-01 14:52:55,976 [lib.cuckoo.core.guest] INFO: Starting analysis on guest (id=01-win7-x64, ip=172.16.254.51)
2016-11-01 14:52:56,980 [lib.cuckoo.core.guest] DEBUG: 01-win7-x64: not ready yet
2016-11-01 14:52:57,984 [lib.cuckoo.core.guest] DEBUG: 01-win7-x64: not ready yet
2016-11-01 14:52:58,978 [lib.cuckoo.core.guest] DEBUG: 01-win7-x64: not ready yet
2016-11-01 14:53:00,983 [lib.cuckoo.core.guest] DEBUG: 01-win7-x64: not ready yet
2016-11-01 14:53:02,038 [lib.cuckoo.core.guest] DEBUG: 01-win7-x64: waiting for status 0x0001
2016-11-01 14:53:02,100 [lib.cuckoo.core.guest] DEBUG: 01-win7-x64: status ready
2016-11-01 14:53:02,324 [lib.cuckoo.core.guest] DEBUG: Uploading analyzer to guest (id=01-win7-x64, ip=172.16.254.51)
2016-11-01 14:53:03,424 [lib.cuckoo.core.guest] DEBUG: 01-win7-x64: analyzer started with PID 2532
2016-11-01 14:53:03,460 [lib.cuckoo.core.guest] DEBUG: 01-win7-x64: waiting for completion
2016-11-01 14:53:03,585 [lib.cuckoo.core.resultserver] DEBUG: LogHandler for live analysis.log initialized.
2016-11-01 14:53:04,474 [lib.cuckoo.core.guest] DEBUG: 01-win7-x64: analysis not completed yet (status=2)
2016-11-01 14:53:04,684 [lib.cuckoo.core.resultserver] DEBUG: File upload request for shots/0001.jpg
2016-11-01 14:53:04,688 [lib.cuckoo.core.resultserver] DEBUG: Uploaded file length: 33668
2016-11-01 14:53:04,688 [lib.cuckoo.core.resultserver] DEBUG: Connection closed: 172.16.254.51:49219
2016-11-01 14:53:05,423 [lib.cuckoo.core.resultserver] DEBUG: File upload request for shots/0002.jpg
2016-11-01 14:53:05,426 [lib.cuckoo.core.resultserver] DEBUG: Uploaded file length: 27164
2016-11-01 14:53:05,426 [lib.cuckoo.core.resultserver] DEBUG: Connection closed: 172.16.254.51:49221
2016-11-01 14:53:05,481 [lib.cuckoo.core.scheduler] ERROR: Analysis failed: The package "modules.packages.exe" start function raised an error: Unable to execute the initial process, analysis aborted.
12 packets captured
12 packets received by filter
0 packets dropped by kernel
2016-11-01 14:53:05,534 [lib.cuckoo.core.plugins] DEBUG: Stopped auxiliary module: Sniffer
2016-11-01 14:53:05,721 [modules.machinery.vsphere] INFO: Powering off virtual machine 01-win7-x64
2016-11-01 14:53:07,066 [lib.cuckoo.core.resultserver] DEBUG: Connection closed: 172.16.254.51:49217
2016-11-01 14:53:07,259 [lib.cuckoo.core.resultserver] DEBUG: Connection closed: 172.16.254.51:49218
2016-11-01 14:53:07,341 [lib.cuckoo.core.scheduler] DEBUG: Released database task #135
2016-11-01 14:53:07,354 [lib.cuckoo.core.plugins] DEBUG: Executed processing module "AnalysisInfo" on analysis at "/opt/cuckoo/storage/analyses/135"
2016-11-01 14:53:07,357 [lib.cuckoo.core.plugins] DEBUG: Executed processing module "BehaviorAnalysis" on analysis at "/opt/cuckoo/storage/analyses/135"
2016-11-01 14:53:07,357 [lib.cuckoo.core.plugins] DEBUG: Executed processing module "DroppedBuffer" on analysis at "/opt/cuckoo/storage/analyses/135"
2016-11-01 14:53:07,359 [lib.cuckoo.core.plugins] DEBUG: Executed processing module "Debug" on analysis at "/opt/cuckoo/storage/analyses/135"
2016-11-01 14:53:07,360 [lib.cuckoo.core.plugins] DEBUG: Executed processing module "Dropped" on analysis at "/opt/cuckoo/storage/analyses/135"
2016-11-01 14:53:07,360 [lib.cuckoo.core.plugins] DEBUG: Executed processing module "ProcessMemory" on analysis at "/opt/cuckoo/storage/analyses/135"
2016-11-01 14:53:08,180 [lib.cuckoo.core.plugins] DEBUG: Executed processing module "Static" on analysis at "/opt/cuckoo/storage/analyses/135"
2016-11-01 14:53:08,242 [lib.cuckoo.core.plugins] DEBUG: Executed processing module "Strings" on analysis at "/opt/cuckoo/storage/analyses/135"
2016-11-01 14:53:08,285 [lib.cuckoo.core.plugins] DEBUG: Executed processing module "TargetInfo" on analysis at "/opt/cuckoo/storage/analyses/135"
2016-11-01 14:53:08,288 [lib.cuckoo.core.plugins] DEBUG: Executed processing module "NetworkAnalysis" on analysis at "/opt/cuckoo/storage/analyses/135"
2016-11-01 14:53:12,997 [lib.cuckoo.core.plugins] DEBUG: Executed processing module "VirusTotal" on analysis at "/opt/cuckoo/storage/analyses/135"
2016-11-01 14:53:12,998 [lib.cuckoo.core.plugins] DEBUG: Executed processing module "TLSMasterSecrets" on analysis at "/opt/cuckoo/storage/analyses/135"
2016-11-01 14:53:13,022 [lib.cuckoo.core.plugins] DEBUG: Running 420 signatures
2016-11-01 14:53:13,201 [lib.cuckoo.core.plugins] DEBUG: Analysis matched signature: antivirus_virustotal
2016-11-01 14:53:13,201 [lib.cuckoo.core.plugins] DEBUG: Analysis matched signature: packer_entropy
2016-11-01 14:53:13,227 [lib.cuckoo.core.plugins] DEBUG: Executed reporting module "JsonDump"
2016-11-01 14:53:13,259 [lib.cuckoo.core.plugins] DEBUG: Executed reporting module "MongoDB"
2016-11-01 14:53:13,700 [lib.cuckoo.core.plugins] DEBUG: Executed reporting module "ReportHTML"
2016-11-01 14:53:13,700 [lib.cuckoo.core.scheduler] INFO: Task #135: reports generation completed (path=/opt/cuckoo/storage/analyses/135)
2016-11-01 14:53:13,744 [lib.cuckoo.core.scheduler] INFO: Task #135: analysis procedure completed
```

Please let me know if you need further details.

Thank you in advance.

**Comment by jbremer** [comment]
Which .NET version do you have installed? Can you share some analyses?

**Comment by overflawed** [comment]
Wow, that was fast!

I have .NET 4.5.2 installed.

Sure, what part of the analysis should I share?

**Comment by jbremer** [comment]
A zip of the whole analysis directory in `storage/analyses/` will do ;) There should also be an `export analysis` button somewhere on the web interface that will make a zip for you.

**Comment by overflawed** [comment]
oops :) Sure, I will provide you in a few minutes.

**Comment by overflawed** [comment]
Actually version 2.0-rc1 doesn't seem to have the "export analysis", but I will zip the analyses/# directory.

**Comment by overflawed** [comment]
I used the following software as a sample and analyzed it in Cuckoo: https://sourceforge.net/projects/notepaddotnet/

analyses/137 folder is attached.

**zip password is:** cuckoo

**Comment by overflawed** [comment]
@jbremer is there anything else I could do to help?
Could you let me know what are you going to look for in the attached zip? I could also try to look it up myself.

By the way, the binary I used to reproduce the error was actually this: https://sourceforge.net/projects/notepaddotnet/files/Notepad%20.NET/Notepad%20.NET%200.2/notepaddotnetsetup02.exe/download
I installed it, and then analyzed the main binary with cuckoo.

**Comment by jbremer** [comment]
Does that binary even run in your Cuckoo VM when you run it manually (or with `free=yes` option)? Don't you think it depends on many 3rd party DLLs that are shipped with Notepad++ that would be unavailable during the analysis?

**Comment by overflawed** [comment]
The very same "high level" error happens when I submit smaller .Net malware binaries that actually run manually in my VM.

I just wanted to make it easier for you to reproduce the error without the need to send a real malware publicly here, but I didn't realize the same "high level error" could be caused by different reasons.

@jbremer  What would be the best place for me to share some files with you in private?
 
Honestly I don't remember if I tried this notepad manually in my VM. I will need to test this very same binary in my environment and get back to you

**Comment by overflawed** [comment]
@jbremer you are right. That particular file (notepaddotnet) depends on some DLLs, I was in a hurry and didn't notice that, sorry for that.

Anyways, my .EXE-only malwares also fails analysis altought they run normally on my VMs.

**Comment by jbremer** [comment]
Can you share one or more of the failing malware samples?

**Comment by overflawed** [comment]
sure, can it be in private?

**Comment by jbremer** [comment]
Of course, feel free to email me at `jbr@cuckoo.sh`

**Comment by GiuseppeLaurenza** [comment]
any news on this problem?

**Comment by overflawed** [comment]
Sorry for the delay. I lost access to this email address.

So, I don't have the infrastructure to test it again :(

But as soon as I have the chance I'll let you know.

**Comment by jbremer** [comment]
This issue has been addressed by @daanfs for our upcoming release. Please stay tuned!

**Comment by jbremer** [comment]
This has been resolved in our `2.0.5` release which we put out just now. Please upgrade and try again! Closing issue, but feel free to reopen if the issue(s) persist for you. Thanks!
Please find also our blogpost https://cuckoosandbox.org/blog/205-office-dde
